### PR TITLE
Remove throwing Get methods from Metadata API

### PR DESCRIPTION
**Pull request by AndriySvyryd**

### CODE REVIEW

File: src/EntityFramework.Relational.Design/RelationalScaffoldingModelFactory.cs [hunk 1]
```diff
@@ -300,12 +300,12 @@ protected virtual IMutableForeignKey VisitForeignKey([NotNull] ModelBuilder mode
 
             try
             {
-                var dependentEntityType = modelBuilder.Model.GetEntityType(GetEntityTypeName(foreignKey.Table));
-                var principalEntityType = modelBuilder.Model.GetEntityType(GetEntityTypeName(foreignKey.PrincipalTable));
+                var dependentEntityType = modelBuilder.Model.FindEntityType(GetEntityTypeName(foreignKey.Table));
+                var principalEntityType = modelBuilder.Model.FindEntityType(GetEntityTypeName(foreignKey.PrincipalTable));
```
Comment (natemcmaster) [inline src/EntityFramework.Relational.Design/RelationalScaffoldingModelFactory.cs]: This is going to lead to NullReferenceExceptions. In this case (and others), the "Get" method doubles as an assertion.

Comment (AndriySvyryd) [inline src/EntityFramework.Relational.Design/RelationalScaffoldingModelFactory.cs]: Yes, we decided that NRE is ok if it's caused by a bug.

Comment (natemcmaster) [inline src/EntityFramework.Relational.Design/RelationalScaffoldingModelFactory.cs]: In this case, it is not a bug, but expected behavior. This is easy to get around by adding a check for null after calling .Find, but it needs to be there for scaffolding to work correctly. Did testing not catch this?

Comment (AndriySvyryd) [inline src/EntityFramework.Relational.Design/RelationalScaffoldingModelFactory.cs]: In that case I'll add null checks and remove code that catches the exceptions.
Tests are still running.

**Comment by natemcmaster** [comment]
:shipit:

**Pull request merged**